include/pinocchio/multibody/joint/joint-composite.hpp

```
@@ -464,6 +485,16 @@ namespace pinocchio
       return res;
     }
 
+    int getJointIndex(const std::string & joint_name) const
+    {
+      auto it = std::find(jointNames.begin(), jointNames.end(), joint_name);
+
+      if (it == jointNames.end())
+        return -1;
```

`JointIndex` is unsigned. So this can't work. `ModelTpl::getJointId` throw an exception if the joint is not found.

---

it doesn't throw when the joint is not found, it sends back the size of the vector of joints from the model. So either i do the same, or i throw. 
Let me know what you think

---

```cpp
    PINOCCHIO_CHECK_INPUT_ARGUMENT(
      (res < INT_MAX), "Id superior to int range. Should never happen.");
```
So this check is not working ? I mean, it was weird…

---

i can't find INT_MAX definition.... but i tried it really quick in python with the sampleManipulator and a fake name, and the function return names.size() (ie 7)

---

Ok, actually, `ModelTpl::getJointId` documentation state that the right behavior…
Anyway, I don't find it's a good one, so we can throw an exception instead.